term_proposal/Term_proposal.md

```diff
@@ -0,0 +1,27 @@
+# XGBoost implementation.
+
+I am Alisher, PhD student, EECS International Graduate Program, National Chiao Tung University.
+
+## Introduction
+In statistics and machine learning, ensemble methods use multiple learning algorithms to obtain better predictive performance than could be obtained from any of the constituent learning algorithms alone. Common types of ensembles: Bayes optimal classifier, Bootstrap aggregating (bagging) and Boosting. This project will be focus on Boosting algorithm. Boosting is a machine learning ensemble meta-algorithm for primarily reducing bias, and also variance in supervised learning, and a family of machine learning algorithms that convert weak learners to strong ones. One of the most popular boosting algorithm among the Kaggle community is XGBoost. 
+
+## Why do I want to do this project?
+In my last Kaggle competition related to an image classification problem, the victory was won by a girl from Japan who used XGBoost algorithm. Even this algorithm is very popular, the most people just try to tune hyperparameters for training the model. But very few people implement XGBoost by their own. The reason is obvious, if a powerful enough tool already exists, then why come up with something new.
+However, I think that for education process is really helpful when you don’t use libraries, but try to implement it by yourself. It can give more insight into how XGBoost works.
+I propose a software project that implements small part of XGBoost algorithm related to regression problem.
+
+## What am I going to do in this project?
+Learn and understand the math background of XGBoost algorithm.
+Implement XGBoost algorithm using linear booster (or if there is enough time, then also gradient booster).
+Plot all results to get better understanding of the XGBoost algorithm.
+Write unit tests.
+Compare to the original XGBoost.
+Prepare final report and presentation.
+
+## References.
+
+- https://www.kaggle.com/
+- https://github.com/dmlc/xgboost
+- https://xgboost.readthedocs.io/en/latest/
```

Comment:
You can analyze the architecture of the existing tool, and state how you'd like to learn and improve it.

Reply:
I am not sure that I can improve the architecture, otherwise it could be one very good paper. I think the main goal of my project is to study and implement this tool based on math background. I am not going to look at the original code, so I will not know how they implemented original XGBoost. I just have the architecture and I will try to use it.

Reply:
Improving the algorithm can probably be a good research paper.  Improving the implementation, on the other hand, often costs one's tenure.  What I meant was the latter.

I did suggest you to take a look at the original code, but it is not a strong opinion.  If this is your life-long work, I'd say you must read as much code as you can.  But in the setting of a term project, use your own judgement.